### PR TITLE
Attempt to fix --dump and --json for AD objects

### LINKED ISSUE #12960: Fix --dump and --json for AD objects
## Rationale

The `--dump` and `--json` options for MOOSE apps output two entries for each AD object, one with a `<RESIDUAL>` postfix and one with a `<JACOBIAN>` postfix. Yet neither version is accepted in the input file, only the name without a (C++ template) postfix is.

## Description

Output only one copy for each AD object syntax - without the template parameter junk. That's what is expected in the input files. The template junk currently breaks auto completion and makes generated Documentation files look awkward.

## Impact

Better syntax output.

### PR DESCRIPTION
**Pull request by tophmatthews**
Close  #12960

### CODE REVIEW

**Comment by tophmatthews** [comment]
Well I'm not sure about the errors yet, but here's an attept at #12960 @dschwen

File: framework/src/parser/Parser.C [hunk 1]
```diff
@@ -824,8 +828,12 @@ Parser::buildFullTree(const std::string & search_string)
           {
             pos = act_name.size();
 
+            // Remove <RESIDUAL> append for AD objects
+            std::string obj_name = moose_obj->first;
+            removeSubstring(obj_name, "<RESIDUAL>");
+
             if (!action_obj_params.collapseSyntaxNesting())
-              name = act_name.substr(0, pos - 1) + moose_obj->first;
+              name = act_name.substr(0, pos - 1) + obj_name;
```
Comment (tophmatthews) [inline framework/src/parser/Parser.C]: Good old trial and error got me to this only substitution. The dump file still has <RESIDUAL>, but autocomplete still works. Hopefully nothing else gets screwed up

**Comment by dschwen** [comment]
Cool! This passes the docs as well (which was a big concern for me). This is great.

**Comment by tophmatthews** [comment]
This and #13098 will hopefully help QOL for AD bigtime!

**Comment by tophmatthews** [comment]
I'm not sure who this would go to for review, @permcody maybe?

**Review by permcody — APPROVED** [review]

**Pull request merged**